woocommerce/woocommerce-ios
# Remove redundant `ProductSelector` initializer, and `supportsMultipleSelection` property.

Closes: #9434
Closes: #9176 

## Description
This PR: 
- Removes the duplicated `ProductSelectorViewModel` initializer, used when we separated logic for single-selection vs multiple-selection.  
- Removes the `supportsMultipleSelection` property from `ProductSelectorViewModel`, and `ProductVariationSelectorViewModel`, as it's always `true`.

## Testing instructions
The initializer is un-used since we launched Product Multi-Selection, the only part that is affected by the changes is product selection state:
- Go to Orders > `+` > `+ Add Products` > Confirm that selecting/unselecting products toggles their selected state (appear as selected/unselected upon tapping)
- Confirm that you can create Orders normally.
